src/jaxsim/parsers/descriptions/link.py

```diff
@@ -58,7 +60,32 @@ def __eq__(self, other: LinkDescription) -> bool:
         if not isinstance(other, LinkDescription):
             return False
 
-        return hash(self) == hash(other)
+        if self.name != other.name:
+            return False
+
+        if not np.allclose(self.mass, other.mass):
+            return False
+
+        if not np.allclose(self.inertia, other.inertia):
+            return False
+
+        if self.index != other.index:
+            return False
+
+        if not np.allclose(self.pose, other.pose):
+            return False
+
+        if self.children != other.children:
+            return False
+
+        # Here only using the name to prevent circular recursion
+        if self.parent is not None and self.parent.name != other.parent.name:
+            return False
+
+        if self.parent is None and other.parent is not None:
+            return False
```

Comment:
Same as above for compactness
```suggestion
        if not all(
            [
                self.name == other.name,
                np.allclose(self.mass, other.mass),
                np.allclose(self.inertia, other.inertia),
                self.index == other.index,
                np.allclose(self.pose, other.pose),
                self.children == other.children,
                (
                    (self.parent is not None and self.parent.name == other.parent.name)
                    if self.parent is not None
                    else other.parent is None
                ),
            ]
        ):
            return False
```